solutions/usaco-599.mdx

```diff
@@ -25,133 +25,165 @@ Bessie's location if the current path is not repeated somewhere else in the map.
 If we were to frame this as a string-matching problem, we would want to find the
 shortest substring of the map such that it only appears once.
 
+To handle angles, we can note that there are only two possible interior angles, $90^\circ$ and $270^\circ$.
+
+![angle pic](usaco-599/di_ang.png)
+
 We use hashing to find the shortest substring of the map. Distinguishing
 vertices from angles, we brute force through lengths in increasing order and use
 a one-base polynomial hash to check if the substring appears only once. If we
 find an appropriate substring, we find the total distance that we travelled
 before finding out where we were.
 
+## Implementation
 **Time Complexity:** $\mathcal{O}(N^3)$
 
+<LanguageSection>
+<CPPSection>
+
 ```cpp
 #include <bits/stdc++.h>
-
 using namespace std;
 
-typedef pair<int, int> pii;
-#define f first
-#define s second
 using ll = long long;
 
-const ll X = 97;
-const ll M = 999999929;
+const ll P = 31;
+const ll MOD = 1e9 + 7;
+const int MX = 2e5;
```

Comment:
don't see a 2e5 anywhere in the problem...can you clarify what this means?

Reply:
okok